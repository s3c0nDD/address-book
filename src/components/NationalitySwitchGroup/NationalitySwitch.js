import React from 'react';

import FormControlLabel from '@material-ui/core/FormControlLabel';
import Switch from '@material-ui/core/Switch';

const NationalitySwitch = ({ checked, handleChange, name }) => {
  const label = name.charAt(0).toUpperCase() + name.substring(1);

  return (
    <FormControlLabel
      control={
        <Switch
          checked={checked}
          onChange={handleChange}
          name={name}
        />
      }
      label={label}
    />
  );
};

export default NationalitySwitch;
