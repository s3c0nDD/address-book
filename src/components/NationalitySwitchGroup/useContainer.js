import { useDispatch, useSelector } from 'react-redux';

import { toggleNationality } from '../../store/module';

const useContainer = () => {
  const dispatch = useDispatch();

  const nationalities = useSelector(state => state.app.nationalities);

  const doToggleNationality = (name) => dispatch(toggleNationality(name));

  return [{ nationalities }, { doToggleNationality }]
};

export default useContainer;