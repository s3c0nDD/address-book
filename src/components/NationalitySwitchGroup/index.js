import NationalitySwitchGroup from './NationalitySwitchGroup';
import NationalitySwitch from './NationalitySwitch';

export default NationalitySwitchGroup;

export {
  NationalitySwitch
}