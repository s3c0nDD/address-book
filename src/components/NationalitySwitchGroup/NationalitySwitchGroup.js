import React from 'react';

import FormControl from '@material-ui/core/FormControl';
import FormHelperText from '@material-ui/core/FormHelperText';
import FormLabel from '@material-ui/core/FormLabel';
import FormGroup from '@material-ui/core/FormGroup';

import useContainer from './useContainer';
import NationalitySwitch from './NationalitySwitch';

const NationalitySwitchGroup = () => {
  const [{ nationalities }, { doToggleNationality }] = useContainer();

  const handleChange = (event) => {
    doToggleNationality(event.target.name);
  };

  return (
    <FormControl component="fieldset">
      <FormLabel component="legend">
        Select nationalities
      </FormLabel>
      <FormGroup>
        {Object.keys(nationalities).map((nationalityName) => (
          <NationalitySwitch
            key={nationalityName}
            checked={nationalities[nationalityName]}
            handleChange={handleChange}
            name={nationalityName}
          />
        ))}
      </FormGroup>
      <FormHelperText
        style={{ maxWidth: 400 }}
      >
        Please choose at least 1 nationality.
        Remember that if you toggle any switch, all
        previously loaded data on main page will be removed.
      </FormHelperText>
    </FormControl>
  );
};

export default NationalitySwitchGroup;
