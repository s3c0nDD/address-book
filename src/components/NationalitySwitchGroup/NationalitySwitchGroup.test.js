import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';

import NationalitySwitch from './NationalitySwitch';
import NationalitySwitchGroup from './NationalitySwitchGroup';

describe('NationalitySwitchGroup', () => {
  let shallow;
  let component;
  let wrapper;
  let toggleMock;

  beforeAll(() => {
    shallow = createShallow();
  });

  beforeEach(() => {
    toggleMock = jest.fn();
    component = (
      <NationalitySwitchGroup
        nationalities={{
          polish: true
        }}
        onToggleNationality={toggleMock}
      />
    );
    wrapper = shallow(component);
  });

  it('shallowly renders and matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('calls change handler on NationalitySwitch click', () => {
    const switchWrapper = wrapper.find(NationalitySwitch);
    const mockedEvent = { target: { name: switchWrapper.props().name }};
    switchWrapper.props().handleChange(mockedEvent);
    expect(toggleMock).toBeCalledTimes(1);
  });
});
