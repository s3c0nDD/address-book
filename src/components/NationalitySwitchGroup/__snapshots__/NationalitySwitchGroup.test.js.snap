// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NationalitySwitchGroup shallowly renders and matches snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  component="fieldset"
>
  <WithStyles(ForwardRef(FormLabel))
    component="legend"
  >
    Select nationalities
  </WithStyles(ForwardRef(FormLabel))>
  <WithStyles(ForwardRef(FormGroup))>
    <NationalitySwitch
      checked={true}
      handleChange={[Function]}
      key="polish"
      name="polish"
    />
  </WithStyles(ForwardRef(FormGroup))>
  <WithStyles(ForwardRef(FormHelperText))
    style={
      Object {
        "maxWidth": 400,
      }
    }
  >
    Please choose at least 1 nationality. Remember that if you toggle any switch, all previously loaded data on main page will be removed.
  </WithStyles(ForwardRef(FormHelperText))>
</WithStyles(ForwardRef(FormControl))>
`;
